lib/fetch-template.js

```diff
@@ -26,35 +26,66 @@ class TemplateError extends Error {
  *
  * @param {string} path
  */
-const readFile = (path) => 
+const readFile = (path) =>
     new Promise((resolve, reject) => {
         fs.readFile(path, 'utf-8', (err, data) => {
             if (err) {
                 reject(new TemplateError(err));
                 return;
-            } 
+            }
             resolve(data);
         });
     });
 
 /**
+ * Returns the template path validating a exactly file or a directory
+ *
+ * @param  {Object} request - Request Object
+ * @param  {String} templatesPath - TemplatesPath config
+ *
+ * @return {Promise} Template Path on success or TemplateError on fail
+ */
+const getTemplatePath = (request, templatesPath) => new Promise((resolve, reject) => {
+    fs.lstat(
+        templatesPath,
+        (err, data) => {
+            if (err) {
+                return reject(new TemplateError(err));
+            }
+
+            if (data.isFile()) {
+                return resolve(templatesPath);
+            }
+            return resolve(path.join(templatesPath, getPathName(request)) + '.html');
+        }
+    );
+});
+
+/**
+ * Returns pathname by request
+ *
+ * @param  {Object} request - Request Object
+ *
+ * @return {String} pathname
+ */
+const getPathName = (request) => url.parse(request.url, true).pathname;
+
+/**
  * Fetches the template from File System
  *
  * @param {string} templatesPath - The path where the templates are stored
  * @param {function=} baseTemplateFn - Function that returns the Base template name for a given page
  */
-module.exports = (templatesPath, baseTemplateFn) => 
+module.exports = (templatesPath, baseTemplateFn) =>
     (request, parseTemplate) => {
-        const pathname = url.parse(request.url, true).pathname;
-        const templatePath = path.join(templatesPath, pathname) + '.html';
-
-        return readFile(templatePath)
+        return getTemplatePath(request, templatesPath)
```

Comment:
@vigneshshanmugam I adjust the duplicated url parsing and refactoring the logic to factory the complete template path. Now I working for break the process in case of single file